ote/src/cljs/ote/views/transit_visualization.cljs

```diff
@@ -694,11 +694,22 @@
            (for [{id :id :as p} previous-packages]
              ^{:key id}
              [:li [pkg p]]))]])]]))
-(defn transit-visualization [e! {:keys [hash->color date->hash loading? highlight service-info
+(defn transit-visualization [e! {:keys [hash->color date->hash route-lines-for-date-loading? route-trips-for-date1-loading?
+                                        route-trips-for-date2-loading?
+                                        route-calendar-hash-loading? route-differences-loading? routes-for-dates-loading?
+                                        service-changes-for-dates-loading?
+                                        highlight service-info
                                         changes selected-route compare open-sections]
-                                 :as transit-visualization}]
+                                 :as   transit-visualization}]
   [:div
-   (when (not loading?)
+   (when (and
```

Comment:
Why does to UI need to know about every loading state? I think it would be cleaner if the events just set one boolean for the UI to check if it is "ready"